### PR TITLE
parser: add a benchmark test for Parse

### PR DESCRIPTION
**Pull request by tiancaiamao**
parser influences the performance of the whole program largly,
we should bench it.

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -989,3 +989,22 @@ func (s *testParserSuite) TestInsertStatementMemoryAllocation(c *C) {
 	runtime.ReadMemStats(&newStats)
 	c.Assert(int(newStats.TotalAlloc-oldStats.TotalAlloc), Less, 1024*500)
 }
+
+func BenchmarkParse(b *testing.B) {
+	var table = []string{
+		"insert into t (c) select c1 from t1 union select c2 from t2",
+		"select c1 from t1 union (select c2 from t2) limit 1, 1",
+		"create table t (c int comment 'comment')",
```
Comment (coocood) [inline parser/parser_test.go]: Why chose those statements?

Comment (shenli) [inline parser/parser_test.go]: Add some statement like "insert into t values (1), (2), (3);". It is used more frequently.

**Comment by shenli** [comment]
LGTM
PTAL @coocood

File: parser/parser_test.go [hunk 2]
```diff
@@ -989,3 +989,23 @@ func (s *testParserSuite) TestInsertStatementMemoryAllocation(c *C) {
 	runtime.ReadMemStats(&newStats)
 	c.Assert(int(newStats.TotalAlloc-oldStats.TotalAlloc), Less, 1024*500)
 }
+
+func BenchmarkParse(b *testing.B) {
+	var table = []string{
+		"create table t (c int)",
```
Comment (coocood) [inline parser/parser_test.go]: We don't need to benchmark `create table`.

**Comment by shenli** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**